crates/agent/src/evolution.rs

```diff
@@ -55,8 +58,8 @@ pub struct EvolvedCollection {
     pub updated_captures: Vec<String>,
 }
 
-fn error_status(err: impl Into<String>) -> anyhow::Result<JobStatus> {
-    Ok(JobStatus::EvolutionFailed { error: err.into() })
+fn error_status(err: impl Into<String>) -> anyhow::Result<Result<JobStatus, CannotAcquireLock>> {
```

Comment:
nit: could you do a dynamic downcast to CannotAcquireLock rather than nesting Result ?

Reply:
done